src/main/java/com/epam/grid/engine/service/JobOperationProviderService.java

```diff
@@ -53,24 +47,20 @@
 public class JobOperationProviderService {
 
     private final String logDir;
-    private final EngineType engineType;
+    private final JobProvider jobProvider;
 
     /**
-     * Collection of providers by engine type.
-     */
-    private Map<EngineType, JobProvider> providers;
-
-    /**
-     * Constructor, sets the specified type of the executed engine and the path to job log.
+     * Constructor, sets created jobProvider bean to the class field and the path to job log.
      *
-     * @param logDir     the path to the directory where all log files will be stored
-     *                   occurred when processing the job
-     * @param engineType an engine for working with jobs
+     * @param jobProvider created JobProvider
+     * @param logDir      the path to the directory where all log files will be stored
+     *                    occurred when processing the job
+     * @see JobProvider
      */
-    public JobOperationProviderService(@Value("${job.log.dir}") final String logDir,
-                                       @Value("${grid.engine.type}") final EngineType engineType) {
+    @Autowired
```

Comment:
As far as I can remember, `@Autowired` is not required for constructors for our spring version. Isn't it?